1.sh

```diff
@@ -1,35 +1,34 @@
 #! /bin/sh
 
-if [-d $1] && [-d $2]
+if [ -d "$1" ] && [ -d "$2" ]
 then
 	echo "Dirs exist"
-elif [! -d $2] 
+elif [ ! -d "$2" ] 
 	echo "Flash path doesn't exist"
-	mkdir $2
+	mkdir "$2"
 fi
 
-for file in $1/*
+for file in "$1"/*
 do
-	echo $(basename $file)
-	flashfile=$($2/$(basename $file))
-	if [! -f $flashfile]; then
-		echo "Copying files to $2"
-		cp $file $2
+	echo "$(basename $file)"
+	flashfile="$2/$(basename $file)"
```

Comment:
I may have misunderstood what you were wanting here, but it looks as though you've accidentally added an additional layer of command substitution (nested command substitution). The variable `flashfile` will now be assigned only the contents of `$2/$(basename $file)".

To break this down:

`$2` will be expanded to the value assigned to the 2nd positional parameter given to your script; the second directory name.

The `$(basename $file)` is command substitution, in its more modern form; it used to be done with graves (```) but that had stylistic and practical issues, so `$()` came to be. You probably figured this out by now, but command substitution allows you to place the output of a command (or even entire blocks of code!) almost anywhere; it allows for a LOT of flexibility.

The `$file` variable is expanded within the command substitution to the value assigned to the variable, which is the currently iterated file in the `for` loop.